### ISSUE TITLE
False positive C0326 bad-whitespace while type annotation and default value for function argument

### ISSUE THREAD

**Issue by XAMeLeOH** [PyCQA/pylint#1421]
### Steps to reproduce
1. Create test file
```
$ cat > mytest.py <<EOF
#!/usr/bin/python3
"""Module docstring"""


def somefunc_annotation(who: str="world") -> str:
    """Say hello"""
    return "Hello " + who
EOF
```
2. run pylint
```
$ pylint \
    '--msg-template={path}:{line}: [{msg_id}({symbol}), {obj}] {msg}' \
    --output-format=text mytest.py
************* Module mytest
mytest.py:4: [C0326(bad-whitespace), ] Exactly one space required around keyword argument assignment
def somefunc_annotation(who: str="world") -> str:
                                ^
```

### Current behavior
Complain that there are spaces required around keyword argument.

### Expected behavior
There should not be an issue in case of simultaneous type annotation and default value for function argument.

### pylint --version output
```
pylint 1.7.0, 
astroid 1.5.1
Python 3.5.2 (default, Jul  5 2016, 12:43:10) 
[GCC 5.4.0 20160609]
```

**Comment by XAMeLeOH** [comment]
According to [PEP 3107](https://www.python.org/dev/peps/pep-3107/#syntax) this issue is invalid, so it should be closed.

**Comment by matt-gardner** [comment]
I don't think PEP 3107 is clear on this issue, actually.  In its example for nested expressions, it gives:

```python
def foo((x1, y1: expression),
        (x2: expression, y2: expression)=(None, None)):
    ...
```
